debug-tools/examples/rectangle-intersection.rs

```diff
@@ -0,0 +1,64 @@
+use embedded_graphics::{
+    pixelcolor::Rgb888,
+    prelude::*,
+    primitives::{PrimitiveStyle, Rectangle},
+};
+use embedded_graphics_simulator::{OutputSettingsBuilder, SimulatorDisplay, Window};
+use framework::prelude::*;
+
+struct RectangleIntersection {
+    top_left: Point,
+    bottom_right: Point,
+}
+
+impl App for RectangleIntersection {
+    type Color = Rgb888;
+    const DISPLAY_SIZE: Size = Size::new(200, 200);
+
+    fn new() -> Self {
+        Self {
+            top_left: Point::new(80, 80),
+            bottom_right: Point::new(150, 150),
+        }
+    }
+
+    fn parameters(&mut self) -> Vec<Parameter> {
+        vec![
+            Parameter::new("top-left", &mut self.top_left),
+            Parameter::new("bottom-right", &mut self.bottom_right),
+        ]
+    }
+
+    fn draw(
+        &self,
+        display: &mut SimulatorDisplay<Self::Color>,
+    ) -> Result<(), std::convert::Infallible> {
+        let base_rectangle = Rectangle::with_corners(Point::new(20, 20), Point::new(100, 100));
+        let moving_rectangle = Rectangle::with_corners(self.top_left, self.bottom_right);
+
+        base_rectangle
+            .into_styled(PrimitiveStyle::with_fill(Rgb888::RED))
+            .draw(display)?;
+
+        moving_rectangle
+            .into_styled(PrimitiveStyle::with_fill(Rgb888::GREEN))
+            .draw(display)?;
+
+        let intersection = base_rectangle.intersection(&moving_rectangle);
+
+        if intersection.size != Size::zero() {
+            intersection
+                .into_styled(PrimitiveStyle::with_fill(Rgb888::BLUE))
+                .draw(display)?;
+        }
```

Comment:
Drawing a zero sized rectangle should be a noop anyway:
```suggestion
        intersection
            .into_styled(PrimitiveStyle::with_fill(Rgb888::BLUE))
            .draw(display)?;
```
But if you want to keep the check for some reason, the preferred way to test for zero sizes is:
```suggestion
        if intersection.is_zero_sized() {
            intersection
                .into_styled(PrimitiveStyle::with_fill(Rgb888::BLUE))
                .draw(display)?;
        }
```

Reply:
Ah yes, it'll be a noop won't it. I've removed the check.